src/uct/ib/base/ib_md.c

```diff
@@ -1168,6 +1172,8 @@ ucs_status_t uct_ib_md_open_common(uct_ib_md_t *md,
                           UCT_MD_FLAG_ADVISE;
     md->reg_cost        = md_config->reg_cost;
     md->relaxed_order   = 0;
+    ucs_strncpy_zero(md->config.gid_ndev, md_config->gid_ndev,
+                     sizeof(md->config.gid_ndev));
```

Comment:
I think it is better to store NULL in case of empty parameter. And then just pass `md->config.gid_ndev` to `uct_ib_iface_init_roce_gid_info`. It will simplify the code a little bit, because you can just call `uct_ib_device_select_gid_by_ndev(dev, port_num, cfg_gid_ndev, &iface->gid_info);`.